pkg/action/actioncontroller.go

```diff
@@ -149,3 +190,25 @@ func (ac *ActionController) createCRD() error {
 	}
 	return k8sutils.CreateCRD(resource, validateCRDInterval, validateCRDTimeout)
 }
+
+// getLatestMigrationStatus returns the a migrationschedule's latest migration's policy type and status
+func getLatestMigrationPolicyAndStatus(migrationSchedule storkv1.MigrationSchedule) (storkv1.SchedulePolicyType, *storkv1.ScheduledMigrationStatus) {
+	var lastMigrationStatus *storkv1.ScheduledMigrationStatus
+	var schedulePolicyType storkv1.SchedulePolicyType
+	for schedulePolicyType, policyMigration := range migrationSchedule.Status.Items {
+		if len(policyMigration) > 0 {
+			lastMigrationStatus = policyMigration[len(policyMigration)-1]
+			return schedulePolicyType, lastMigrationStatus
+		}
+	}
+	return schedulePolicyType, lastMigrationStatus
```

Comment:
MigrationScheduleStatus.Items is a map[SchedulePolicyType][]*ScheduledMigrationStatus. 

Because migrationschedule supports all the different types of policies the status can have the key of any policy type. ( I agree that it will be mostly a single policy type but still items will be a map).
The idea is to get for any key whose entries are more than one (that means thats the policy type getting used) as the policyKey and among those pick out the last appended one which is the latest. 

The similar looping also gets used in migrationController code.